video/out/w32_common.c

```diff
@@ -1435,7 +1435,7 @@ static int gui_thread_control(struct vo_w32_state *w32, int request, void *arg)
         if (!w32->taskbar_list3 || !w32->tbtnCreated)
             return VO_TRUE;
 
-        if (!pstate->playing) {
+        if (!pstate->playing || !pstate->taskbar_progress) {
```

Comment:
Well, it's calling `ITaskbarList3_SetProgressState` whenever progress state is changed.

When progress is hidden (player is in idle mode or taskbar-progress has been disabled) `ITaskbarList3_SetProgressState` will be called with `TBPF_NOPROGRESS` flag, which will reset the button to its original state without progress bar and reset progress value to 0 (there is no need to call `ITaskbarList3_SetProgressValue` in this case).

Otherwise `ITaskbarList3_SetProgressValue` will be called to set percentage on the progress bar and then it will be shown with either `TBPF_NORMAL` or `TBPF_PAUSED` flag by `ITaskbarList3_SetProgressState`.

Or am I missing the point?